src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1299,5 +1306,115 @@ public Task CompactDatabase()
+                                if (timer.ElapsedMilliseconds > 1000)
+                                {
+                                    Onprogress(progress);
+                                    progress.Progress = string.Empty;
+                                    timer.Restart();
+                                }
+                                ThrowIfTimeout(configuration.Timeout, timeout);
+
+                            }
+                            progress.Progress += $"{await process.StandardOutput.ReadToEndAsync()}{Environment.NewLine}";
+                            Onprogress(progress);
+                            if (process.ExitCode != 0)
+                            {
+                                throw new ApplicationException($"The data export tool have exited with code {process.ExitCode}.");
+                            }
+
+                            if (File.Exists(configuration.OutputFilePath) == false)
+                            {
+                                throw new FileNotFoundException($"Was expecting the output file to be located at {configuration.OutputFilePath}, but it is not there.");
+                            }
+
+                            using (database.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                            using (var reader = File.OpenRead(configuration.OutputFilePath))
+                            using (var stream = new GZipStream(reader, CompressionMode.Decompress))
+                            {
+                                var source = new StreamSource(stream, context);
+                                var destination = new DatabaseDestination(database);                                
+                                var smuggler = new DatabaseSmuggler(database, source, destination, database.Time, result: result,onProgress: Onprogress);
+
+                                smuggler.Execute();
+                                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                                {
+                                    var json = result.ToJson();
+                                    context.Write(writer, json);
+                                }
+                            }
+                        }
+                        catch (Exception e)
+                        {
+                            result.AddError(e.ToString());
+                            throw e;
+                        }                        
+                        return (IOperationResult)result;
+                    });
+                }, operationId, token);
+
+            void ThrowIfTimeout(TimeSpan? timespan, Stopwatch t)
+            {
+                if (timespan.HasValue && t.Elapsed > timespan.Value)
+                {
+                    throw new TimeoutException($"Export is taking more than the configured timeout {timespan.Value}");
```

Comment:
for me this message is like warning, not exception 
Export is taking more than the configured timeout (but I will continue)

Reply:
Yes, I think we should let the user decide on actual cancelling

Reply:
The user can cancel the operation from the on going operation in the studio. There is no way for me to get user feedback from this code path, is there?

Reply:
The user can cancel the operation from the on going operation in the studio. There is no way for me to get user feedback from this code path, is there?

Reply:
Yes, you don't need to do anything except send an on progress message with the warning